### ISSUE TITLE
Handle early exits in FOR loops

### ISSUE THREAD

**Issue by jomag** [jomag/bajsic#10]
Early exit from a FOR-loop, like in the following example, are problematic:

```vb
10 PRINT "HELLO"
20 FOR I=0 TO 10
30 GOTO 10
40 NEXT I
```

As each FOR statement will push data to the stack, which the NEXT statement will pop, this example will eventually run out of memory as the NEXT statement is never reached, and a new FOR statement data is pushed to the stack for every loop.

This is a quite hard problem to solve, as it's perfectly legal to exit a FOR-loop, and the NEXT-statement could still be reached after that happens:

```vb
10 PRINT "HELLO"
20 FOR I=0 TO 10
30 GOTO 50
35 PRINT "WORLD"
40 NEXT I
50 GOTO 35
```

This wikipage has some documentation of how one BASIC dialect (Microsoft BASIC as used on C64) handles it: https://www.c64-wiki.com/wiki/NEXT

In short:

- When the FOR-NEXT loop terminates normally, the information is removed from the return stack.
- When a program encounters a NEXT statement, all inner-loops are terminated and their information removed from the return stack.
- All FOR-NEXT loops commenced within a subroutine are terminated and their information removed from the return stack when the program encounters a RETURN statement.
- If a FOR statement is encountered, any existing loop using the same variable name along with any subsequent unfinished FOR-NEXT loops are terminated and their information removed from the return stack.
